code/modules/projectiles/guns/projectile.dm

```diff
@@ -19,6 +19,7 @@
 	var/obj/item/ammo_storage/magazine/stored_magazine = null
 	var/obj/item/ammo_casing/chambered = null
 	var/mag_type = ""
+	var mag_type_restricted = "" //better magazine manipulation
```

Comment:
bad dont make this a string make this a path, also why not make this a list of paths?